pkg/flare/flare.go

```diff
@@ -0,0 +1,435 @@
+			}
+		}
+	}
+
+	return nil
+}
+
+func extractDateFromPlainTextLine(line, logDateLayout string) (time.Time, error) {
+	idxs := plainTextLogDateRegex.FindIndex([]byte(line))
+	if idxs == nil {
+		return time.Time{}, ErrDateNotFound
+	}
+
+	strDate := line[idxs[0]+1 : idxs[1]-1]
+	date, err := time.Parse(logDateLayout, strDate)
+	if err != nil {
+		return time.Time{}, err
+	}
+	return date, nil
+}
+
+func extractDateFromJSONLine(line, logDateLayout string) (time.Time, error) {
+	logPropsMap := make(map[string](interface{}))
+	err := json.Unmarshal([]byte(line), &logPropsMap)
+	if err != nil {
+		return time.Time{}, err
+	}
+	lineDateStr, ok := logPropsMap["time"].(string)
+	if !ok {
+		return time.Time{}, ErrExtractDateFromJSONLogLine
+	}
+	lineDate, err := time.Parse(logDateLayout, lineDateStr)
+	if err != nil {
+		return time.Time{}, err
+	}
+	return lineDate, nil
+}
+
+func (f *Flare) handleLogLine(line string, start, end *time.Time) (string, error) {
+	lineDate, err := dateExpractors[f.logFormat](line, f.LogDateLayout)
+	if err != nil {
+		return "", err
+	}
+	if (start == nil || lineDate.Equal(*start) || lineDate.After(*start)) &&
+		(end == nil || lineDate.Equal(*end) || lineDate.Before(*end)) {
+		redactedLine, err := f.redactSecrets(line)
+		if err != nil {
+			return "", err
+		}
+		return redactedLine, nil
+	}
+	return "", nil
+}
+
+func (f *Flare) processLogFile(inputFileName, outputPath string, startDate, endDate *time.Time) (retErr error) {
+	skippedLines := 0
+	sourceFile, err := os.Open(inputFileName)
+	if err != nil {
+		return err
+	}
+	defer sourceFile.Close()
+
+	outputFileName := fmt.Sprintf("%s%s", outputPath, inputFileName)
+	outputFile, err := os.OpenFile(outputFileName, os.O_WRONLY|os.O_CREATE, FilePremissions)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if e := outputFile.Close(); retErr == nil {
+			retErr = e
+		}
+	}()
+
+	bw := bufio.NewWriter(outputFile)
+	defer func() {
+		if e := bw.Flush(); retErr == nil {
+			retErr = e
+		}
+	}()
+
+	first := true
+	scanner := bufio.NewScanner(sourceFile)
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		pLine, err := f.handleLogLine(line, startDate, endDate)
+		if err != nil {
+			skippedLines += 1
+			fmt.Println("failed to process log line: ", err)
+		}
+
+		if pLine != "" {
+			b := strings.Builder{}
+			if !first {
+				b.WriteString("\n")
+			}
+			b.WriteString(line)
+			if _, err := bw.WriteString(b.String()); err != nil {
+				return err
+			}
+		}
+
+		first = false
+	}
+
+	fmt.Printf("Skipped %d log lines\n", skippedLines)
+	return nil
+}
+
+func isDirectory(path string) (bool, error) {
+	fileInfo, err := os.Stat(path)
+	if err != nil {
+		return false, err
+	}
+
+	return fileInfo.IsDir(), err
+}
+
+func (f *Flare) redactSecrets(line string) (string, error) {
+	return redactSecrets(line, f.secretReplacementValue)
+}
+
+func redactSecrets(line string, secretReplacementValue string) (string, error) {
+	detectedSecrets, err := secretScanner.Scan(line)
```

Comment:
Perhaps I didn't track the flow of execution correctly.  I don't understand how this can work for lakeFS configuration files.  For instance `auth.encrypt.secret_key` is a random string whose format we do not control.  It can also appear in environment variables.  It is a really sensitive secret!

Similarly, I think blockstore.signing.secret_key is not _actually_ secret (ask @N-o-Z), but we still probably don't want it to go out.

I believe that the easy solution for config is to use our existing facilities to prevent outputting secret values from config - there are clearly-marked types here.

Reply:
`redactSecrets` isn't used for the config. For the config, we use `SecureString` to redact secrets